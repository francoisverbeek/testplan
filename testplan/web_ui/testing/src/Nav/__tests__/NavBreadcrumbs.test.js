import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import NavBreadcrumbs from '../NavBreadcrumbs';

function defaultProps() {
  const entry = {
    uid: '123',
    name: 'test',
    status: 'passed',
    type: 'testplan',
    case_count: {
      passed: 0,
      failed: 0,
    },
  };
  return {
    entries: [entry],
    handleNavClick: jest.fn(),
  };
}

describe('NavBreadcrumbs', () => {
  let props;
  let mountedNavBreadcrumbs;
  const renderNavBreadcrumbs = () => {
    if (!mountedNavBreadcrumbs) {
      mountedNavBreadcrumbs = shallow(
        <NavBreadcrumbs {...props} />
      );
    }
    return mountedNavBreadcrumbs;
  };

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = defaultProps();
    mountedNavBreadcrumbs = undefined;
  });

  afterEach(() => {
    // Resume style injection once test is finished.
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
    props.handleNavClick.mockClear();
  });

  it('shallow renders without crashing', () => {
    renderNavBreadcrumbs();
  });

  it('shallow renders the correct HTML structure', () => {
    const navBreadcrumbs = renderNavBreadcrumbs();
    expect(navBreadcrumbs).toMatchSnapshot();
  });

  it('calls handleNavClick when nav entry has been clicked', () => {
    const handleNavClick = props.handleNavClick;
    const navBreadcrumbs = renderNavBreadcrumbs();

    navBreadcrumbs.find('li').simulate('click');
    expect(handleNavClick.mock.calls.length).toEqual(1);
  })

});