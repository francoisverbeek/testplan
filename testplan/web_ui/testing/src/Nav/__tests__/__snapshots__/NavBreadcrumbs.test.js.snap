// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBreadcrumbs shallow renders the correct HTML structure 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBreadcrumbs
    entries={
        Array [
            Object {
              "case_count": Object {
                "failed": 0,
                "passed": 0,
              },
              "name": "test",
              "status": "passed",
              "type": "testplan",
              "uid": "123",
            },
          ]
    }
    handleNavClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="breadcrumbContainer_1f5xahc"
>
        <li
                onClick={[Function]}
        >
                <div
                        className="breadcrumbEntry_d724hk"
                >
                        <NavEntry
                                caseCountFailed={0}
                                caseCountPassed={0}
                                name="test"
                                status="passed"
                                type={undefined}
                        />
                </div>
        </li>
</ul>,
      "className": "navBreadcrumbs_qyzuci",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            onClick={[Function]}
>
            <div
                        className="breadcrumbEntry_d724hk"
            >
                        <NavEntry
                                    caseCountFailed={0}
                                    caseCountPassed={0}
                                    name="test"
                                    status="passed"
                                    type={undefined}
                        />
            </div>
</li>,
        ],
        "className": "breadcrumbContainer_1f5xahc",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "123",
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="breadcrumbEntry_d724hk"
>
              <NavEntry
                            caseCountFailed={0}
                            caseCountPassed={0}
                            name="test"
                            status="passed"
                            type={undefined}
              />
</div>,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavEntry
                caseCountFailed={0}
                caseCountPassed={0}
                name="test"
                status="passed"
                type={undefined}
/>,
              "className": "breadcrumbEntry_d724hk",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "caseCountFailed": 0,
                "caseCountPassed": 0,
                "name": "test",
                "status": "passed",
                "type": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="breadcrumbContainer_1f5xahc"
>
          <li
                    onClick={[Function]}
          >
                    <div
                              className="breadcrumbEntry_d724hk"
                    >
                              <NavEntry
                                        caseCountFailed={0}
                                        caseCountPassed={0}
                                        name="test"
                                        status="passed"
                                        type={undefined}
                              />
                    </div>
          </li>
</ul>,
        "className": "navBreadcrumbs_qyzuci",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              onClick={[Function]}
>
              <div
                            className="breadcrumbEntry_d724hk"
              >
                            <NavEntry
                                          caseCountFailed={0}
                                          caseCountPassed={0}
                                          name="test"
                                          status="passed"
                                          type={undefined}
                            />
              </div>
</li>,
          ],
          "className": "breadcrumbContainer_1f5xahc",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "123",
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="breadcrumbEntry_d724hk"
>
                <NavEntry
                                caseCountFailed={0}
                                caseCountPassed={0}
                                name="test"
                                status="passed"
                                type={undefined}
                />
</div>,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavEntry
                  caseCountFailed={0}
                  caseCountPassed={0}
                  name="test"
                  status="passed"
                  type={undefined}
/>,
                "className": "breadcrumbEntry_d724hk",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "caseCountFailed": 0,
                  "caseCountPassed": 0,
                  "name": "test",
                  "status": "passed",
                  "type": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
